authors/apps/authentication/serializers.py

```diff
@@ -26,8 +29,23 @@ class Meta:
         # or response, including fields specified explicitly above.
         fields = ['email', 'username', 'password', 'token']
 
+    
+
+
     def create(self, validated_data):
         # Use the `create_user` method we wrote earlier to create a new user.
+        password = validated_data.get('password') #we get a key from the dictionary validated data
+        pattern = re.compile(
+                r"^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,128}$"
+            )
+        if not pattern.match((password)):
+            raise serializers.ValidationError(
+                {
+                    "password" :[
+                    "password must contain a numeric character"
+                    ]
+                }
+                )
         return User.objects.create_user(**validated_data)
```

Comment:
Thanks, @NicholusMuwonge, for the work you could as well add an extra validator to check for a username.

Reply:
@NicholusMuwonge, @reifred's request was not adressed.
Please do so.

Reply:
I have added the username validator @reifred